doc/manuals/orion-api.md

```diff
@@ -2295,6 +2326,676 @@ Some considerations to take into account when using custom notifications:
+
+results in
+
+```
+3
+```
+
+#### toFixed
+
+Rounds a number to a number of decimal digits.
+
+Extra arguments: number of decimal digits.
+
+Example (being context `{"c": 3.18}`):
+
+```
+c|toFixed(1)
+```
+
+results in
+
+```
+3.2
+```
+
+#### log
+
+Return the natural logarithm of a given number
+
+Extra arguments: none.
+
+Example (being context `{"c": 3.14}`):
+
+```
+c|log
+```
+
+results in
+
+```
+1.144222799920162
+```
+
+#### log10
+
+Return the base 10 logarithm of a given number
+
+Extra arguments: none.
+
+Example (being context `{"c": 3.14}`):
+
+```
+c|log10
+```
+
+results in
+
+```
+0.49692964807321494
+```
+
+#### log2
+
+Return the base 2 logarithm of a given number
+
+Extra arguments: none.
+
+Example (being context `{"c": 3.14}`):
+
+```
+c|log2
+```
+
+results in
+
+```
+1.6507645591169025
+```
+
+#### sqrt
+
+Return the square root of a given number
+
+Extra arguments: none.
+
+Example (being context `{"c": 3.14}`):
+
+```
+c|sqrt
+```
+
+results in
+
+```
+1.772004514666935
+```
+
+#### replaceStr
+
+Replace occurrences of a string with another in the input string.
+
+Extra arguments:
+* Source string to replace
+* Destination string to replace
+
+Example (being context `{"c": "foobar"}`):
+
+```
+c|replaceStr('o','u')
+```
+
+results in
+
+```
+"fuubar"
+```
+
+#### mapper
+
+Returns a value among several choices based in one to one mapping. This function is based in an array of *values* and an array of *choices* (which length is exactly the same). Thus, if the input value is equal to the *i*-th item of *values*, then *i*-th item of *choices* is returned.
+
+This transformation returns `null` if some problem with the arguments is found (i.e. input is not found among the values, choices length is not exacly the same as values, the input is not an string, etc.)
+
+Extra arguments:
+* values array
+* choices array
+
+Example (being context `{"c": "fr", "values": ["es", "fr", "de"], "choices": ["Spain", "France", "Germany"]}`):
+
+```
+c|mapper(values,choices)
+```
+
+results in
+
+```
+"France"
+```
+
+#### thMapper
+
+Returns a value among several choices based in threshold values. This function is based in an array of *values* and an array of *choices* (which length is exactly the same as values plus one). Thus, if the input value is between the *i*-th and the *i+1*-th item of *values*, then *i*+1-th item of *choices* is returned.
+
+This transformation returns `null` if some problem with the arguments is found (i.e. choices length is not exacly the same as values plus one, some of the items in the values array is not a number, etc.)
+
+Extra arguments:
+* values array
+* choices array
+
+Example (being context `{"c": 0.5, "values": [-1, 1], "choices": ["low", "medium", "high"]}`):
+
+```
+c|thMapper(values,choices)
+```
+
+results in
+
+```
+"medium"
+```
+
+### Failsafe cases
+
+As failsafe behaviour, evaluation returns `null` in the following cases:
+
+* Some of the transformation used in the expression is unknown
```

Comment:
Probably, is a good idea to add an example:

```suggestion
* Some of the transformation used in the expression is unknown (e.g. `A|undefinedExpression`)
```

Reply:
Good idea!

Fixed in [06cbe4e](https://github.com/telefonicaid/fiware-orion/pull/4511/commits/06cbe4e3506c56a67c81207a86cd29555dd58fdf)